{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "28740545-b9bc-4f04-a37e-919e552a92f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('DHFR', 'Major', 'S108N')\n",
      "('PfCRT', 'Minor', 'N75K')\n",
      "('PfCRT', 'Minor', 'M343K')\n",
      "('PfCRT', 'Major', 'N75K')\n",
      "('DHFR', 'Minor', 'C59R')\n",
      "('DHFR', 'Major', 'C59R')\n",
      "('PfCRT', 'Major', 'K76T')\n",
      "('PfCRT', 'Minor', 'K76T')\n",
      "('PfCRT', 'Major', 'I356T')\n",
      "('PfCRT', 'Minor', 'I356T')\n",
      "('DHPS_437Corrected ', 'Major', 'A437G')\n",
      "('DHPS_437Corrected ', 'Major', 'S436A')\n",
      "('PfCRT', 'Major', 'Q271E')\n",
      "('DHPS_437Corrected ', 'Minor', 'S436A')\n",
      "('DHPS_437Corrected ', 'Minor', 'K540E')\n",
      "('PfCRT', 'Minor', 'Q271E')\n",
      "('DHPS_437Corrected ', 'Major', 'K540E')\n",
      "('DHPS_437Corrected ', 'Minor', 'A437G')\n",
      "('DHPS_437Corrected ', 'Major', 'A581G')\n",
      "('PfMDR1 ', 'Minor', 'N86D')\n",
      "('DHPS_437Corrected ', 'Minor', 'A581G')\n",
      "('PfCRT', 'Major', 'N75E')\n",
      "('PfCRT', 'Minor', 'N75E')\n",
      "('DHFR', 'Major', 'N51I')\n",
      "('PfCRT', 'Major', 'M74I')\n",
      "('PfMDR1 ', 'Major', 'Y184F')\n",
      "('PfCRT', 'Minor', 'M74L')\n",
      "('PfCRT', 'Major', 'A220S')\n",
      "('PfCRT', 'Minor', 'A220S')\n",
      "('PfCRT', 'Minor', 'R371I')\n",
      "('PfCRT', 'Minor', 'M74I')\n",
      "('PfMDR1 ', 'Minor', 'Y184F')\n",
      "('PfMDR1 ', 'Minor', 'N86Y')\n",
      "('DHPS_437Corrected ', 'Minor', 'A581V')\n",
      "('DHPS_437Corrected ', 'Minor', 'A613V')\n",
      "('DHPS_437Corrected ', 'Major', 'A613S')\n",
      "('PfMDR1 ', 'Major', 'N86Y')\n",
      "('DHPS_437Corrected ', 'Minor', 'S436F')\n",
      "('PfCRT', 'Major', 'R371I')\n",
      "('PfMDR1 ', 'Major', 'D1246Y')\n",
      "('DHPS_437Corrected ', 'Minor', 'A613S')\n",
      "('PfCRT', 'Wildtype', 'C72S')\n",
      "('PfCRT', 'Wildtype', 'T93S')\n",
      "('PfCRT', 'Wildtype', 'G353V')\n",
      "('PfCRT', 'Wildtype', 'N75E')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'A581G')\n",
      "('PfCRT', 'Wildtype', 'V73V')\n",
      "('PfMDR1 ', 'Wildtype', 'Y184F')\n",
      "('PfMDR1 ', 'Wildtype', 'D1246Y')\n",
      "('PfCRT', 'Wildtype', 'Q271E')\n",
      "('PfMDR1 ', 'Wildtype', 'N1042D')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'K540E')\n",
      "('PfMDR1 ', 'Wildtype', 'S1034C')\n",
      "('PfCRT', 'Wildtype', 'H97Y')\n",
      "('PfCRT', 'Wildtype', 'R371I')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'A613S')\n",
      "('PfCRT', 'Wildtype', 'M74I')\n",
      "('PfCRT', 'Wildtype', 'K76T')\n",
      "('PfMDR1 ', 'Wildtype', 'N86Y')\n",
      "('K13', 'Wildtype', 'N537I')\n",
      "('K13', 'Wildtype', 'G538V')\n",
      "('K13', 'Wildtype', 'R539T')\n",
      "('K13', 'Wildtype', 'I543T')\n",
      "('K13', 'Wildtype', 'A675V')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'S436A')\n",
      "('PfCRT', 'Wildtype', 'I356T')\n",
      "('PfCRT', 'Wildtype', 'N326S')\n",
      "('DHFR', 'Wildtype', 'I164L')\n",
      "('mitochondrial genome - CYTB CDS', 'Wildtype', 'I258M')\n",
      "('mitochondrial genome - CYTB CDS', 'Wildtype', 'Y268N')\n",
      "('K13', 'Wildtype', 'Y493H')\n",
      "('DHFR', 'Wildtype', 'C59R')\n",
      "('K13', 'Wildtype', 'P441L')\n",
      "('K13', 'Wildtype', 'F446I')\n",
      "('K13', 'Wildtype', 'G449A')\n",
      "('K13', 'Wildtype', 'P527H')\n",
      "('K13', 'Wildtype', 'P553L')\n",
      "('K13', 'Wildtype', 'R561H')\n",
      "('K13', 'Wildtype', 'V568G')\n",
      "('K13', 'Wildtype', 'P574L')\n",
      "('K13', 'Wildtype', 'A578S')\n",
      "('K13', 'Wildtype', 'C580Y')\n",
      "('K13', 'Wildtype', 'D584V')\n",
      "('K13', 'Wildtype', 'R622I')\n",
      "('PfCRT', 'Wildtype', 'C350R')\n",
      "('PfCRT', 'Wildtype', 'A220S')\n",
      "('PfCRT', 'Wildtype', 'M343L')\n",
      "('PfCRT', 'Wildtype', 'F145I')\n",
      "('DHFR', 'Wildtype', 'A16V')\n",
      "('K13', 'Wildtype', 'R515K')\n",
      "('K13', 'Wildtype', 'C469F')\n",
      "('K13', 'Wildtype', 'C469Y')\n",
      "('K13', 'Wildtype', 'M476I')\n",
      "('K13', 'Wildtype', 'A481V')\n",
      "('K13', 'Wildtype', 'N458Y')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'A437G')\n",
      "('DHFR', 'Wildtype', 'N51I')\n",
      "('PfCRT', 'Wildtype', 'N75K')\n",
      "('PfCRT', 'Wildtype', 'M343K')\n",
      "('PfMDR1 ', 'Wildtype', 'N86D')\n",
      "('PfCRT', 'Wildtype', 'M74L')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'A581V')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'A613V')\n",
      "('DHPS_437Corrected ', 'Wildtype', 'S436F')\n"
     ]
    }
   ],
   "source": [
    "with open (\"test2.csv\", \"r\") as t1: ###Open the document created from dataframe workflow\n",
    "    dict_SNPs={} ###Calculate major, minor, and wildtypes by going through each line\n",
    "    for lines in t1: ##Add number of major, minor, \n",
    "        currentVAF=0 ##These are sum for individaul and pools with all SNP annotation\n",
    "        #print(lines)\n",
    "        if \"Sequence Name\" not in lines: \n",
    "            if \"->\" in lines.split(\",\")[8]: ###Check if there are arrows and change of VAF due to difference of reads\n",
    "                currentVAF=float(lines.split(\",\")[8][0:4].strip(\"%\")) ###Assign the left side of the VAF when there is arrow\n",
    "            else:\n",
    "                currentVAF=float(lines.split(\",\")[8].strip(\"%\")) ###If there is no arrow just strip percentage and assign VAF\n",
    "            if currentVAF >=50 and \"individual\" in lines and \"21\" not in lines: ##If VAF is larger or same as 50 then VAF is assigned as major\n",
    "                #print(lines.split(\",\"))\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]) not in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]]=1.0\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]) in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]]=dict_SNPs[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]]+1.0\n",
    "            if 0<currentVAF<50 and \"individual\" in lines and \"21\" not in lines: ###If VAF is smaller than 50 but larger than 0 than VAF is assigned as minor\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]) not in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]]=1.0\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]) in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]]=dict_SNPs[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]]+1.0\n",
    "            if currentVAF ==0 and \"individual\" in lines and \"21\" not in lines: ## If VAF is 0 then it is assigned as wildtype\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]) not in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]]=1.0\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]) in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]]=dict_SNPs[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]]+1.0\n",
    "            if currentVAF >=50 and \"pooled\" in lines and \"21\" not in lines: ##If VAF is larger or same as 50 then VAF is assigned as major\n",
    "                #print(lines.split(\",\"))\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]) not in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]]=float(lines.split(\",\")[-2])\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]) in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]]=dict_SNPs[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6]]+float(lines.split(\",\")[-2])\n",
    "            if 0<currentVAF<50 and \"pooled\" in lines and \"21\" not in lines: ###If VAF is smaller than 50 but larger than 0 than VAF is assigned as minor\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]) not in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]]=float(lines.split(\",\")[-2])\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]) in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]]=dict_SNPs[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6]]+float(lines.split(\",\")[-2])\n",
    "            if currentVAF ==0 and \"pooled\" in lines and \"21\" not in lines: ## If VAF is 0 then it is assigned as wildtype\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]) not in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]]=float(lines.split(\",\")[-2])\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]) in dict_SNPs:\n",
    "                    dict_SNPs[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]]=dict_SNPs[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6]]+float(lines.split(\",\")[-2])\n",
    "\n",
    "\n",
    "with open (\"test2.csv\", \"r\") as t1: ###Open the document created from dataframe workflow\n",
    "    dict_SNPs_minus={} ###Calculate major, minor, and wildtypes by going through each line\n",
    "    for lines in t1: ##Add number of major, minor, \n",
    "        currentVAF=0 ##These are sum for individaul and pools with all SNP annotation\n",
    "        #print(lines)\n",
    "        if \"Sequence Name\" not in lines: \n",
    "            if \"->\" in lines.split(\",\")[8]: ###Check if there are arrows and change of VAF due to difference of reads\n",
    "                currentVAF=float(lines.split(\",\")[8][0:4].strip(\"%\")) ###Assign the left side of the VAF when there is arrow\n",
    "            else:\n",
    "                currentVAF=float(lines.split(\",\")[8].strip(\"%\")) ###If there is no arrow just strip percentage and assign VAF\n",
    "            if currentVAF >=50 and \"individual\" in lines and \"21\" not in lines: ##If VAF is larger or same as 50 then VAF is assigned as major\n",
    "                #print(lines.split(\",\"))\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]) not in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]]=1.0\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]) in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]]=dict_SNPs_minus[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]]+1.0\n",
    "            if 0<currentVAF<50 and \"individual\" in lines and \"21\" not in lines: ###If VAF is smaller than 50 but larger than 0 than VAF is assigned as minor\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]) not in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]]=1.0\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]) in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]]=dict_SNPs_minus[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]]+1.0\n",
    "            if currentVAF ==0 and \"individual\" in lines and \"21\" not in lines: ## If VAF is 0 then it is assigned as wildtype\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]) not in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]]=1.0\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]) in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]]=dict_SNPs_minus[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]]+1.0\n",
    "            if currentVAF >=50 and \"pooled\" in lines and \"21\" not in lines: ##If VAF is larger or same as 50 then VAF is assigned as major\n",
    "                #print(lines.split(\",\"))\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]) not in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]]=float(lines.split(\",\")[-2])\n",
    "                if (lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]) in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]]=dict_SNPs_minus[lines.split(\",\")[5],\"Major\",lines.split(\",\")[6][0:-1]]+float(lines.split(\",\")[-2])\n",
    "            if 0<currentVAF<50 and \"pooled\" in lines and \"21\" not in lines: ###If VAF is smaller than 50 but larger than 0 than VAF is assigned as minor\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]) not in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]]=float(lines.split(\",\")[-2])\n",
    "                if (lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]) in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]]=dict_SNPs_minus[lines.split(\",\")[5],\"Minor\",lines.split(\",\")[6][0:-1]]+float(lines.split(\",\")[-2])\n",
    "            if currentVAF ==0 and \"pooled\" in lines and \"21\" not in lines: ## If VAF is 0 then it is assigned as wildtype\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]) not in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]]=float(lines.split(\",\")[-2])\n",
    "                if (lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]) in dict_SNPs_minus:\n",
    "                    dict_SNPs_minus[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]]=dict_SNPs_minus[lines.split(\",\")[5],\"Wildtype\",lines.split(\",\")[6][0:-1]]+float(lines.split(\",\")[-2])\n",
    "#print(dict_SNPs)\n",
    "dict_SNPs_SUM={} ###Find sum of major, minor, wildtypes, total  samples\n",
    "\n",
    "dict_SNPs_2={} ##The purpose of adding wildtype to second mutations\n",
    "for items in dict_SNPs:\n",
    "    #print(items)\n",
    "    if (items[0], \"Wildtype\", items[2]) not in dict_SNPs: ##Check if wildtype is not in the dictionary \n",
    "        if (items[0],\"Wildtype\",items[2][0:-1]) in dict_SNPs_minus:\n",
    "            dict_SNPs_2[items[0], \"Wildtype\", items[2]]=dict_SNPs_minus[items[0],\"Wildtype\",items[2][0:-1]] ###Assign wildtype to the SNP change without wildtype information\n",
    "\n",
    "#print(dict_SNPs_2)\n",
    "for items in dict_SNPs_2: ##Add back to the original dictionary\n",
    "    #print(items)\n",
    "    dict_SNPs[items]=dict_SNPs_2[items]\n",
    "\n",
    "for items in dict_SNPs: ##Create a beginng value of dictionary to later add up to total\n",
    "    dict_SNPs_SUM[items[0],items[2]]=0\n",
    "    \n",
    "###Figure out why sum  is weird for seaborn table why is it differet from WVAF total calculation so much?\n",
    "\n",
    "#dict_count={} ## dictionary for couting how many times dict_SNP_minus which is without last letter of G_Annotation is added.\n",
    "for items in dict_SNPs: ##First check if the item is in dictionary otherwise change the value of dict_count\n",
    "    #print(items)\n",
    "    dict_SNPs_SUM[items[0],items[2]]=dict_SNPs_SUM[items[0],items[2]]+dict_SNPs[items]\n",
    "\n",
    "#print(dict_SNPs_SUM)        \n",
    "dict_SNPS_ratio_sum = {} ##Find the ratio of major, minor, wildtypes in total samples\n",
    "\n",
    "for items in dict_SNPs:\n",
    "    #print(items)\n",
    "    dict_SNPS_ratio_sum[items[1],items[2]]=0\n",
    "\n",
    "for items in dict_SNPs:\n",
    "    #print(items)\n",
    "    dict_SNPS_ratio_sum[items[1],items[2]]=dict_SNPS_ratio_sum[items[1],items[2]]+round(dict_SNPs[items]/dict_SNPs_SUM[items[0],items[2]],5)\n",
    "\n",
    "#print(dict_SNPS_ratio_sum)\n",
    "dict_SNPs_final = {} ##Create final form for the SNPs\n",
    "for items in dict_SNPs:\n",
    "    #print(items)\n",
    "    if (items[0],items[2]) not in dict_SNPs_final:\n",
    "        dict_SNPs_final[items[0],items[2]]=[0,0,0,0]\n",
    "    if items[1]==\"Wildtype\":\n",
    "        dict_SNPs_final[items[0],items[2]][0]=dict_SNPS_ratio_sum[items[1],items[2]]\n",
    "    if items[1]==\"Major\":\n",
    "        dict_SNPs_final[items[0],items[2]][1]=dict_SNPS_ratio_sum[items[1],items[2]]\n",
    "    if items[1]==\"Minor\":\n",
    "        dict_SNPs_final[items[0],items[2]][2]=dict_SNPS_ratio_sum[items[1],items[2]]\n",
    "    dict_SNPs_final[items[0],items[2]][3]=dict_SNPs_SUM[items[0],items[2]]\n",
    "\n",
    "#for items in dict_SNPs_final:\n",
    "    #print(dict_SNPs_final[items])\n",
    "with open (\"Tab_Table_SNP_Total1.csv\", \"w\") as t2:\n",
    "    t2.write(\"Gene+SNP,Wildtype: AF=0%,Minor: AF < 50%,Major: AF >= 50%,Samplesize\\n\")\n",
    "    for items in dict_SNPs_final:\n",
    "        if items[0]==\"DHPS_457Corrected \":\n",
    "            t2.write(\"DHPS\"+\":\"+items[1]+\",\"+str(dict_SNPs_final[items][0])+\",\"+str(dict_SNPs_final[items][1])+\",\"+str(dict_SNPs_final[items][2])+\",\"+\"N=\"+str(dict_SNPs_final[items][3])+\"\\n\")\n",
    "        if items[0]==\"mitochondrial genome - CYTB CDS\":\n",
    "            t2.write(\"CYTB\"+\":\"+items[1]+\",\"+str(dict_SNPs_final[items][0])+\",\"+str(dict_SNPs_final[items][1])+\",\"+str(dict_SNPs_final[items][2])+\",\"+\"N=\"+str(dict_SNPs_final[items][3])+\"\\n\")         \n",
    "        if items[0]!= \"DHPS_457Corrected \" and items[0]!=\"mitochondrial genome - CYTB CDS\":\n",
    "            t2.write(items[0]+\":\"+items[1]+\",\"+str(dict_SNPs_final[items][0])+\",\"+str(dict_SNPs_final[items][1])+\",\"+str(dict_SNPs_final[items][2])+\",\"+\"N=\"+str(dict_SNPs_final[items][3])+\"\\n\")\n",
    "    \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc833220-ad8d-44cd-bccf-d9fd8801f491",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
